cachito/workers/pkg_managers/pip.py

```diff
@@ -1514,3 +1517,101 @@ def upload_raw_package(repo_name, artifact_path, dest_dir, filename, is_request_
     to_nexus_hoster = not is_request_repository
     log.debug("Uploading %r as a raw package to the %r Nexus repository", artifact_path, repo_name)
     nexus.upload_raw_component(repo_name, dest_dir, components, to_nexus_hoster)
+
+
+def generate_pip_conf_content(nexus_repo_identifier, username, password, custom_ca_path=None):
+    """
+    Generate a pip.conf file with the nexus repo url and authentication configured.
+
+    :param str nexus_repo_identifier: the nexus repo identifier,
+        e.g. cachito-pip-hosted-<request_id>
+    :param str username: the username of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str password: the password of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str custom_ca_path: the path to set ``cert`` to in the pip.conf file; if not provided,
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
+    # https://help.sonatype.com/repomanager3/formats/npm-registry#npmRegistry-AuthenticationUsingBasicAuth
+    # Create the request specific nexus repo url to be passed as `index-url` within pip.conf
+    nexus_repo_url = base64.b64encode(
+        f"https://[{username}[:{password}]@]{nexus_repo_identifier}".encode("utf-8")
```

Comment:
`nexus_repo_identifier` is just the repo name, right? This need to be the full URL, something like (not exactly) `https://username:password@localhost:8081/repository/cachito-pip-hosted-1/simple`.

The square brackets around username and password should also be removed, they are usually used in documentation to indicate that something is optional.

Reply:
Oops, thanks for noticing the brackets! I forgot to remove them after copying from documentation. I planned on discussing the exact URL needed. @athos-ribeiro Do you know the exact URL for request specific nexus hosted repository ?

Reply:
This is the `cachito_nexus_url` value in the `cachito/workers/config.py` for the hosted repositories. It would be the hoster url for the proxy and the permanent raw repository.